### DISCUSSION TITLE
Datadog Tracing. How to get the TraceId and SpanId reported to datadog?

### DISCUSSION THREAD

**Discussion by golfingal72** [apollographql/router#1459] [Q&A]
I am configuring the datadog plugin in the router.   I would like to get access to the datadog Trace_ID and Span_ID for the requests.  Is there a way to do this in a plugin?

**Comment by bnjjj** [comment]
If you want to access to the TraceId and SpanId created on datadog when sending your traces you won't be able because basically it's sending the trace to the datadog agent which itself sent the traces to the datadog API. So the trace_id you can fetch is the one created by `tracing` the rust library. What's your use case ?

BTW I highly recommend to use OTLP exporter and plug the datadog agent to the OTLP exporter, it's better supported. [Here you have documentation on datadog to use OTLP exporter ](https://docs.datadoghq.com/fr/tracing/trace_collection/open_standards/otel_collector_datadog_exporter/)and [here you have documentation to plug tracing to OTLP exporter in the router](https://www.apollographql.com/docs/router/configuration/tracing#opentelemetry-collector-via-otlp)

**Reply by golfingal72** [reply]
@bnjjj 
Thanks for the info... i will look at your links...  I am writing a "tomcat"like access log and want to trap the trace_id and span_id in them...

I was planning on using in the router:

telemetry:
  tracing:
    datadog:
      # Either 'default' or a URL
      endpoint: default

We have a datadog sidecar running in our containers that will listen to the default port.

I actually figured out how to get the numbers:

use opentelemetry::global;
use opentelemetry::trace::TraceContextExt;
use tracing::Span;
use tracing_opentelemetry::OpenTelemetrySpanExt;

                let cx = &Span::current().context();
                let span = cx.span();
                let span_context = span.span_context();
                let span_id = format!("{:x}", span_context.span_id());
                let trace_id = format!("{:x}", span_context.trace_id());
                let telemetry_trace_id = u128::from_str_radix(&trace_id, 16); // this is a Result<u128, Error>
                let telemetry_span_id = u128::from_str_radix(&span_id, 16); // this is a Result<u128, Error>

**Reply by bnjjj** [reply]
Is it the same IDs than the ones used in datadog ? I mean Datadog agent doesn't create their own IDs ? Just out of curiosity
